sympy/functions/special/bessel.py

```diff
@@ -223,11 +223,22 @@ def _eval_rewrite_as_jn(self, nu, z, **kwargs):
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
         nu, z = self.args
-        arg = z.as_leading_term(x)
-        if x in arg.free_symbols:
-            return arg**nu/(2**nu*gamma(nu + 1))
-        else:
-            return self.func(nu, z.subs(x, 0))
+        try:
+            arg = z.as_leading_term(x)
+        except NotImplementedError:
+            return self
+        _, e = arg.as_coeff_exponent(x)
+
+        if e.is_positive:
+            if x in arg.free_symbols:
```

Comment:
Can e be positive if x is not in arg.free_symbols?

Reply:
No it can't , I thought something like leading terms calls of `besselj(1, 1/y)` or `besselj(1,  y)` would end up having `e` as `1 ` for some reason but I see it will be 0 , which means I need to change the order of evaluation a bit .